scripts/verify-correctness.js

```diff
@@ -35,6 +39,15 @@ module.exports = async callback => {
     const auctionElementsDecoded = await getOrdersPaginated(BatchExchange, 100)
     const bracketTraderAddresses = argv.brackets.map(address => address.toLowerCase())
 
+    // fetch all token infos(decimals, symbols etc) and prices upfront for the following verification
+    const globalPriceStorage = {}
+    const relevantOrders = auctionElementsDecoded.filter(order => bracketTraderAddresses.includes(order.user.toLowerCase()))
+    relevantOrders.forEach(async order => {
+      const tokenInfo = await fetchTokenInfoFromExchange([order.sellToken, order.buyToken])
+      await getDexagPrice(tokenInfo[order.sellToken].symbol, tokenInfo[order.sellToken].symbol, globalPriceStorage)
+      await getDexagPrice(tokenInfo[order.sellToken].symbol, "USDC", globalPriceStorage)
+    })
```

Comment:
Is it? Because we are storing every result in globalPriceStorage and globalTokenPromisesFromId, I thought that it would be efficient. Where do we lose time?

Reply:
Whoops, you are right, I thought that `.forEach` executed each function one after the other but it executes everything in parallel. So there isn't much to gain by changing this code, sorry about that.